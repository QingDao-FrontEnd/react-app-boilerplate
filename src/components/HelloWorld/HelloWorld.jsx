/* @flow */

import React, { Component } from 'react';
import './HelloWorld.less';

export type State = {
  name: string
}

export type Actions = {
  setName: Function
}

type Props = State|Actions;

export default class HelloWorld extends Component {
  constructor (props: Props): void {
    super (props);
  }
  handleNameChange = (event: any): void => {
    this.props.setName (event.target.value);
  }
  render (): ?React$Element<any> {
    return <div>
      <h2>Hello, {this.props.name}!</h2>
      <input
        value={this.props.name}
        onChange={this.handleNameChange}
      />
    </div>;
  }
}
