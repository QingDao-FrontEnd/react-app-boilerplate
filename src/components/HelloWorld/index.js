/* @flow */

import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import HelloWorld from "./HelloWorld";
import { setName } from "~/reducers/name";

function mapStateToProps (state: any): Object {
  return {
    name: state.name
  };
}

function mapDispatchToProps (dispatch: any): Object {
  return bindActionCreators (
    {
      setName
    },
    dispatch
  );
}

export default connect (mapStateToProps, mapDispatchToProps) (HelloWorld);
