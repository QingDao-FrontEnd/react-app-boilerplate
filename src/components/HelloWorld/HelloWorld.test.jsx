import React, { Component } from 'react';
import { HelloWorld } from './HelloWorld';
import renderer from 'react-test-renderer';

describe ('HelloWorld', () => {
  test ('component should mount', () => {
    const component = renderer.create (
      <HelloWorld name="Bob" />
    );
    let tree = component.toJSON ();
    expect (tree).toMatchSnapshot ();
  });
});
