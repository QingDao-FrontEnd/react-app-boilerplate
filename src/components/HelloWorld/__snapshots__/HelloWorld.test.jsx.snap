exports[`HelloWorld component should mount 1`] = `
<div>
  <h2>
    Hello, 
    Bob
    !
  </h2>
  <input
    onChange={[Function]}
    value="Bob" />
</div>
`;
